### PR TITLE
[Term Entry] Dart List .addAll()

### LINKED ISSUE #4300: [Term Entry] Dart List .addAll()
### Reason/inspiration (optional)

We would like a new term entry under Dart [List](https://www.codecademy.com/resources/docs/dart/concept/list) for the Dart List `.addAll()` method. 
Provide the entry in a markdown file under `docs/content/dart/concept/list/terms/addAll/addAll.md`

The entry should
    - Have an introductory paragraph about what the method is and does.
    - A syntax section with a pseudo code block with an explanation of it.
    - An example showing the method in use.

Please refer to the [concept entry templates](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by manviii27**
### Description

- Created a new term entry under Dart List .addAll() method

### Issue Solved

- Closes #4300 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [ ] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/dart/concepts/list/terms/addAll/addAll.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+---
+
+The **`.addAll()`** method in Dart which allows you to append multiple elements from one list to another list. This method is useful when you want to combine the contents of multiple lists into a single list. This method does not create a new list rather modifies the original list.
+
+## Syntax
+
+```pseudo
+list1.addAll(list2);
+```
+
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/addAll/addAll.md:24]: Describe the list1 and list2 parameters just below the syntax block. It should be in this format - 
```
- `paramter`: it's description.
```

File: content/dart/concepts/list/terms/addAll/addAll.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+Tags:
+  - 'Dart'
+  - 'Arithmetic'
+  - 'Lists'
+CatalogContent:
+  - 'learn-dart'
+  - 'paths/computer-science'
+---
+
+The **`.addAll()`** method in Dart which allows you to append multiple elements from one list to another list. This method is useful when you want to combine the contents of multiple lists into a single list. This method does not create a new list rather modifies the original list.
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/addAll/addAll.md:17]:
```suggestion
The **`.addAll()`** method in Dart allows you to append multiple elements from one list to another. This method is useful when you want to combine the contents of multiple lists into a single list. This method does not create a new list but rather modifies the original list.

```

File: content/dart/concepts/list/terms/addAll/addAll.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+## Syntax
+
+```pseudo
+list1.addAll(list2);
+```
+
+## Example
+
+The `.addAll()` method only works on a growable list. If the list is a fixed-length list, then it throws an `Unsupported operation` error and retains all objects in the list.
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/addAll/addAll.md:27]: You can add this sentence as a note here and replace it with the actual description of the code.

File: content/dart/concepts/list/terms/addAll/addAll.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
+---
+Title: '.addAll()'
+Description: 'Adds multiple objects from a list.'
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/addAll/addAll.md:3]:
```suggestion
Description: 'Adds multiple elements from a list.'

```

**Review by avdhoottt — CHANGES_REQUESTED** [review]
Hey, @manviii27, Excellent entry. Please make the following changes.

**Comment by manviii27** [comment]
@avdhoottt I've made the changes, please review

File: content/dart/concepts/list/terms/addAll/addAll.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+Tags:
+  - 'Dart'
+  - 'Arithmetic'
+  - 'Lists'
+CatalogContent:
+  - 'learn-dart'
+  - 'paths/computer-science'
+---
+
+The **`.addAll()`** method in Dart allows you to append multiple elements from one list to another. This method is useful when you want to combine the contents of multiple lists into a single list. This method does not create a new list but rather modifies the original list.
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/addAll/addAll.md:17]:
```suggestion
The **`.addAll()`** method in Dart allows appending multiple elements from one list to another. This method is useful to combine the contents of multiple lists into a single list. This method does not create a new list but rather modifies the original list.

```

File: content/dart/concepts/list/terms/addAll/addAll.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+```
+
+- `list1`: The list in which another list is to be added.
+- `list2`: The list to be added to list1.
+
+## Example
+
+This code demonstrates the usage of the `addAll()` method in Dart. It appends different lists to existing lists. The first example appends a list `[4, 5, 6]` to an existing list `nums`. The second example combines the lists `otherNumbers` and `extraNumbers` and appends them to the `numbers` list. The third example appends a list `additionalElements` to a fixed length list `nonGrowableList`.
+
+**Note** - The `.addAll()` method only works on a growable list. If the list is a fixed-length list, it throws an `Unsupported operation` error and retains all objects.
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/addAll/addAll.md:33]:
```suggestion
> **Note** - The `.addAll()` method only works on a growable list. If the list is a fixed-length list, it throws an `Unsupported operation` error and retains all objects.

```

File: content/dart/concepts/list/terms/addAll/addAll.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+```pseudo
+list1.addAll(list2);
+```
+
+- `list1`: The list in which another list is to be added.
+- `list2`: The list to be added to list1.
+
+## Example
+
+This code demonstrates the usage of the `addAll()` method in Dart. It appends different lists to existing lists. The first example appends a list `[4, 5, 6]` to an existing list `nums`. The second example combines the lists `otherNumbers` and `extraNumbers` and appends them to the `numbers` list. The third example appends a list `additionalElements` to a fixed length list `nonGrowableList`.
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/addAll/addAll.md:31]:
```suggestion
This code demonstrates the usage of the `.addAll()` method in Dart. It appends different lists to existing lists. The first example appends a list `[4, 5, 6]` to an existing list `nums`. The second example combines the lists `otherNumbers` and `extraNumbers` and appends them to the `numbers` list. The third example appends a list `additionalElements` to a fixed length list `nonGrowableList`.

```

**Review by avdhoottt — APPROVED** [review]
Thanks for making the changes. PR is ready for the second review.

**Review by Sriparno08 — APPROVED** [review]
Looks good 👍

**Pull request merged**